### ISSUE TITLE
Workflow for multiple different experiments.

### ISSUE THREAD

**Issue by briankosw** [facebookresearch/hydra#940]
I wanted to learn more about the best workflow for when working on multiple different kinds of experiments. Let's say I have configurations A and B, where A and B will have overlapping hyperparameter types but might have different hyperparameter types (e.g. configuration A doesn't use Dropout but configuration B has a Dropout probability). Would I just feed the train script with the corresponding configuration as in `python main.py --config configA` instead of using `@hydra.main(configA)` in the script? Would I create multiple train scrips, each with its corresponding configuration using `@hydra.main(configA)`?

**Comment by omry** [comment]
This sounds like a perfect use case for the primary feature of Hydra, which is config composition.
Please take a look at the basic tutorial, and in particular at [config groups](https://hydra.cc/docs/tutorials/basic/your_first_app/config_groups).

Generally speaking, composition allow you to access an infinite number of configs. you should not normally need to override your primary config - just to compose something different from it.
